compiler/test/runnable/memberOfProperty.d

```diff
@@ -0,0 +1,41 @@
+extern(C) void printf(const char*, ...);
+
+extern(C) void main() {
+    int varB = :max;
+    Context varS = :max;
+    Enum varE = :End;
+    Enum varF1 = functionReturn();
+    Enum varF2 = identity(:Middle);
+    printf("%d <> %d <> %d <> %d <> %d\n", varB, varS.value, varE, varF1, varF2);
+
+    switch(varF1) {
+        case :Start:
+            static assert(!__traits(compiles, { const fail = :max; }));
+            break;
+        default:
+            assert(0);
+    }
+
+    static assert(!__traits(compiles, { const fail = :max; }));
```

Comment:
should also check for `auto`, `immutable`, static`, `shared`,  `__gshared` (any other type redundant modifier I'm forgetting?). 
Also should check for initialisation at 
* module scope, 
* class/struct scope
* (other scopes I may be forgetting?)